### ISSUE TITLE
Assert when using navigation and mouse input simultaneously

### ISSUE THREAD

**Issue by jadwallis** [ocornut/imgui#1867]
**Version/Branch of Dear ImGui:**

Commit 9165280 (v1.62 WIP, master)

**My Issue/Question:** It seems to be possible to cause ImGui to fail an assert when using navigation mode (I think either keyboard or gamepad, but I've only tested properly with keyboard) with ImGuiConfigFlags_NavEnableSetMousePos enabled, if the user tries to use by both the mouse input and navigation input simultaneously under certain conditions. As far as I understand, it should be valid to have both navigation and mouse position setting enabled at the same time, even if using both mouse and keyboard input simultaneously isn't exactly a 'sensible' use case?

I've managed to repro this fairly consistently first in our own codebase, and then using the provided ImGui demo (see repro steps below), when interacting with the main menu bar. It might also occur for other widgets, but I don't have a repro for that.

I'm not sure when this first starting occurring - I've only hit this now because I'm upgrading our codebase's (somewhat out of date) version of ImGui to latest.

**Standalone, minimal, complete and verifiable example:**

- Run the ImGui Demo (any version should do, but I've been using the DX11 demo).
- In the Demo window, expand the 'Inputs, Navigation & Focus' tab.
- Enable the NavEnableKeyboard and NavEnableSetMousePos flags.
- Open the Examples menu from the main menu bar.
- Use just the mouse to move up and down the menu (just move, don't click anything), and observe this is all fine.
- Then use just the keyboard to move up and down the menu (with the arrow keys), and observe this is all fine.
- Now do both mouse and keyboard navigation simultaneously. You may need to 'fuzz' it a bit by moving up and down quickly and repeatedly. This should trigger an assert!

The assert in question is: `!g.NavDisableHighlight && g.NavDisableMouseHover` which can be found at line 3262 of imgui.cpp (at commit 9165280).

The callstack is:
```
ImGui::NavUpdate() Line 3262
ImGui::NewFrame() Line 3740
ImGui_ImplDX11_NewFrame() Line 683
main(int __formal, char * * __formal) Line 169
```

Thanks for your help. :)

**Comment by ocornut** [comment]
Hello James,

Thanks for your report. I looked at it and confirmed that given the circumstance, removing the assert is appropriate so I just did that. The last mouse moment which disable `g.NavDisableMouseHover` will win over the `SetMousePos` request since the call so the mouse will keep its position.

(Not 100% sure I guess the reason it happens mostly in the menu is because Selectable by default update the navigation position and it is easy to transition from mouse to keyboard when browsing menus (Windows does that too)).

Out of curiosity, why are you enabling `io.NavEnableSetMousePos` ? Especially since you showing a DX11 callstack here. I think it mostly makes sense as a visibility aid when using a TV and not much else (I even semi considered obsoleting the feature).

-Omar
